### PR TITLE
Set up workflow for golangci

### PR DESCRIPTION
**Pull request by mmorel-35**
This PR doesn't fix the identified problem raised by the analysis

### CODE REVIEW

**Comment by gmlewis** [comment]
What prompted you to write this PR?

**Comment by mmorel-35** [comment]
Hi! I saw it defined in https://github.com/google/go-querystring, I thought the project might be interested

**Comment by gmlewis** [comment]
@willnorris - I defer to your judgment on this one and #2019.

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+linters:
+  enable:
+    - dogsled
```
Comment (obalunenko) [inline .golangci.yml:3]: suggest to add ercheck and govet to the list
https://golangci-lint.run/usage/linters/

Comment (mmorel-35) [reply .golangci.yml:3]: Those two are active by default, see https://golangci-lint.run/usage/linters/#enabled-by-default-linters. I haven't deactivated them

File: .github/workflows/linter.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      matrix:
+        go-version: [1.x]
+        platform: [ubuntu-latest]
+    runs-on: ${{ matrix.platform }}
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: golangci-lint
+      uses: golangci/golangci-lint-action@v2
```
Comment (willnorris) [inline .github/workflows/linter.yml:16]:
```suggestion
      uses: golangci/golangci-lint-action@5c56cd6c9dc07901af25baab6f2b0d9f3b7c3018 #v2.5.2
```

Comment (willnorris) [reply .github/workflows/linter.yml:16]: Google requires using exact SHAs for third-party actions.

**Review by willnorris — CHANGES_REQUESTED** [review]

File: .github/workflows/linter.yml (hunk 2 above)
Comment (mmorel-35) [reply .github/workflows/linter.yml]: Done :)!

**Review by willnorris — APPROVED** [review]

**Comment by willnorris** [comment]
so [that's a lot of errors](https://github.com/google/go-github/runs/3364021858?check_suite_focus=true#step:3:67).  I see that you have golangci-lint returning a 0 status, so this doesn't break builds.  What's the plan then... a handful of follow-up PRs that address these individual errors, and then we eventually remove the `--issues-exit-code=0`?  The other approach I've taken in the past is to just disable the failing linters, and then in the commits that fix the errors, also re-enable the linter.

We might also end up wanting to disable some of these, or at least for certain paths.  For example, the dogsled errors in test files are probably not worth fixing.  Same for the duplicate code detection.  Our tests do duplicate some code and that's fine.  Trying to refactor to remove the duplication is likely to make the tests less readable (at least in some cases).

**Comment by mmorel-35** [comment]
The --issues-exit-code=0 is to display the actual triggered warnings/errors without making the job failing indeed.
Then it's entirely up to you to decide what to do, if you want to remove a l'inter or just some rules indeed

**Comment by willnorris** [comment]
What about something like:

``` yaml
linters:
  # TODO: fix errors so that all of the linters below pass.
  # The linters that are commented out, as well as those explicitly disabled,
  # are currently failing.  We should fix those failures or define exclusion
  # rules, and then enable those linters.
  enable:
    - dogsled
    - dupl
    - gofmt
    - goimports
  # - gosec
    - misspell
    - nakedret
  # - stylecheck
  # - unconvert
  # - unparam
  # - whitespace
  disable:
    - errcheck
    - gosimple
    - staticcheck
    - ineffassign
    - unused
issues: 
  exclude-rules:
  - linters:
    - dogsled
    text:  "declaration has 3 blank identifiers"
    path: _test\.go
  - linters:
    - dupl
    path: _test\.go
```

then, as folks want to fix these, they can enable a specific linter locally, run golangci-lint to see what's failing, and fix those errors.  That doesn't give us visibility in the CI logs of what is failing, but I think that's okay for now, especially since these won't take very long to fix.

**Pull request merged**